### ISSUE TITLE
Unit tests fail due to switch to Parquet

### ISSUE THREAD

**Issue by PGijsbers** [openml/openml-python#1157]
I wanted to investigate which unit tests fail and why (and how to get them working again).
One of the changes that broke multiple tests was the switch from ARFF files to Parquet. For example, a dataset which had a boolean value in ARFF would have be a categorical type with two allowed values - in parquet they simply are a pandas boolean type. 
In another example, see below, the Parquet had stored numeric data as float64 but our tests expect them to be uint8 (because there are no missing values and casting the column to uint8 loses no data).

![image](https://user-images.githubusercontent.com/15890747/183911587-c377a96d-07d0-4dee-a907-07672a7a87c8.png)

How do we proceed with this? I propose updating the unit tests/expected behavior where reasonable (for example, integrating the `bool` dtype instead of holding on to a two-valued categorical). Sometimes it _might_ make sense to instead expect changes to the parquet file (as the example in the image), but at the same time I don't know if the parquet readers in all languages can deal with the different types.

_side note: I am not entirely sure why the data is stored in float64 when AFAIK the openml-python module was used to convert the data.._

**Comment by mfeurer** [comment]
Hi,
I deciding this on a case-by-case basis makes sense. Answers about the existing cases:
* How do we know that these should be bools and not categoricals? Since the data was uploaded as arff, we can only guess this I assume. Also, this somewhat changes the OpenML dataset definition by allowing more data types, so it really needs to be defined on an OpenML level before we can implement something in Python.
* I think @amueller implemented the uint8 data type casting to save space when pickling a numpy array on disk. If we no longer do that, and depending on how much caching is still needed, we might be able to drop this downcasting and always use a commonly used format (float64)

**Comment by PGijsbers** [comment]
> Since the data was uploaded as arff, we can only guess this I assume. 

Yes, but we already converted the stringy `"True"` and `"False"` to boolean `True` and `False`. 

> and depending on how much caching is still needed

I'll look into that

**Comment by mfeurer** [comment]
> Yes, but we already converted the stringy "True" and "False" to boolean True and False.

You mean in Python? If that's the case, we should continue doing this I guess.

**Comment by sebffischer** [comment]
It would be great this whole conversion could be properly documented properly somewhere, because it makes it somewhat tricky to migrate. 

Problems I have encountered are: 
1. different data types such as the character --> bool conversion. I guess this cannot be avoided but maybe it would be good to make a summary somewhere where we can document all those conversions that are happening
2. different column names which really should not happen I think

**Comment by PGijsbers** [comment]
To be clear, are you talking specifically about how `openml-python`  handles `ARFF` to `pandas.DataFrame`? Or about how the conversion of `ARFF` to `parquet`  was done server-side?

**Comment by sebffischer** [comment]
I was talking about the conversion from arff to parquet that was done on the server side, not the openml-python package.
I just mentioned it here because you already had a related discussion going on.  

I think the naming bug was on the side of the R api, but i found a case where a column seems to have disappeared in the parquet version of a dataset: https://github.com/openml/OpenML/issues/1165

**Comment by SubhadityaMukherjee** [comment]
I guess this one is fixed already? @PGijsbers

**Comment by PGijsbers** [comment]
The tests are green, in that sense this issue is resolved.
The new parquet files are generated with an updated process that uses similar optimizations that openml-python employed, truncating to int8, boolean, and so on. 
Documentation of this process should be transparent, but that's not for this repository, opened an issue there.
